includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -798,6 +799,31 @@ public function detach_payment_method( $payment_method_id ) {
 		);
 	}
 
+	/**
+	 * Records a new Terms of Service agreement.
+	 *
+	 * @param string $source     A string, which describes where the merchant agreed to the terms.
+	 * @param string $user_name  The user_login of the current user.
+	 * @param string $user_ip    IP address of the current user.
+	 * @param string $user_agent User agent string for the current user.
+	 *
+	 * @return bool A flag, which indicates whether the agreement was saved.
+	 *
+	 * @throws API_Exception If an error occurs.
+	 */
+	public function add_tos_agreement( $source, $user_name, $user_ip, $user_agent ) {
```

Comment:
Would it be better to pass in an array here? We need to keep PHP8 compat in mind though.

Reply:
I don't think so. The list of arguments is not that long and not used in many places, I think this is much clearer.  Also, this way it's consistent, see the `create_and_confirm_intention` for example: All parameters are actual parameters. I don't think converting to an array and checking individual values afterwards makes sense.